Sources/Core/Shared/ComposedOperation.swift

```diff
@@ -8,26 +8,28 @@
 
 import Foundation
 
-public class ComposedOperation<T: NSOperation>: Operation, OperationDidFinishObserver {
+public class ComposedOperation<T: NSOperation>: Operation {
 
     public let target: Operation
     public var operation: T
 
-    public convenience init(_ operation: T) {
-        self.init(operation: operation)
+    public convenience init(_ op: T) {
+        self.init(operation: op)
     }
 
-    init(operation composed: T) {
-        target = composed as? Operation ?? GroupOperation(operations: [composed])
-        operation = composed
+    init(operation op: T) {
+        self.target = op as? Operation ?? GroupOperation(operations: [op])
+        self.operation = op
         super.init()
         name = "Composed Operation"
-        target.name = "Composed <\(T.self)>"
-        target.addObserver(self)
+        target.name = target.name ?? "Composed <\(T.self)>"
```

Comment:
Unfortunately this means that there is now a failing test in `ReachableOperationTest`.